src/types/event-types.ts

```diff
@@ -1,8 +1,22 @@
 import type { Event } from './event-log.js';
 import type { GenericMessageReply } from '../core/message-reply.js';
+import type { RecordsFilter } from './records-types.js';
 import type { AuthorizationModel, GenericMessage } from './message-types.js';
 import type { DwnInterfaceName, DwnMethodName } from '../enums/dwn-interface-method.js';
 
+/**
+ * @param method optional array of methods to filter.
+ * @param interface optional array of interfaces to filter.
+ * @param author optional author to filter.
+ *
+ * @param watermark optional watermark of the last message received from this filter.
+ */
+export type EventsFilter = RecordsFilter & {
+  method?: string[];
+  interface?: string[];
+  author?: string;
```

Comment:
The documentation should be above the actual properties for auto-complete/hint to work in IDEs.

```suggestion
  /** optional author to filter. */
  author?: string;
```